python/pyarrow/types.pxi

```diff
@@ -72,11 +74,17 @@ cdef class DataType:
     def __repr__(self):
         return '{0.__class__.__name__}({0})'.format(self)
 
-    def __richcmp__(DataType self, DataType other, int op):
+    def __richcmp__(DataType self, object other, int op):
+        cdef DataType other_type
+        if not isinstance(other, DataType):
```

Comment:
This should probably check for `basestring`/`str` only and raise otherwise to prevent any weird red herring errors from comparisons returning a meaningless `False` value.

Reply:
done